include/k4a/k4a.h

```diff
@@ -1203,7 +1203,57 @@ K4A_EXPORT k4a_buffer_result_t k4a_device_get_serialnum(k4a_device_t device_hand
  */
 K4A_EXPORT k4a_result_t k4a_device_get_version(k4a_device_t device_handle, k4a_hardware_version_t *version);
 
-/** Get the Azure Kinect color sensor control value.
+/** Get the Azure Kinect color sensor control capabilities.
+ *
+ * \param device_handle
+ * Handle obtained by k4a_device_open().
+ *
+ * \param command
+ * Color sensor control command.
+ *
+ * \param support_auto
+ * Location to store whether the color sensor's control support auto mode or not.
+ * true if it supports auto mode, otherwise false.
+ *
+ * \param min_value
+ * Location to store the color sensor's control minimum value of /p command.
+ *
+ * \param max_value
+ * Location to store the color sensor's control maximum value of /p command.
+ *
+ * \param step_value
+ * Location to store the color sensor's control step value of /p command.
+ *
+ * \param default_value
+ * Location to store the color sensor's control default value of /p command.
+ *
+ * \param default_mode
+ * Location to store the color sensor's control default mode of /p command.
+ *
+ * \returns
+ * ::K4A_RESULT_SUCCEEDED if the value was successfully returned, ::K4A_RESULT_FAILED if an error occurred
+ *
+ * \relates k4a_device_t
+ *
+ * \xmlonly
+ * <requirements>
+ *   <requirement name="Header">k4a.h (include k4a/k4a.h)</requirement>
+ *   <requirement name="Library">k4a.lib</requirement>
+ *   <requirement name="DLL">k4a.dll</requirement>
+ * </requirements>
+ * \endxmlonly
+ */
+K4A_EXPORT k4a_result_t k4a_device_get_color_control_capabilities(k4a_device_t device_handle,
+                                                                  k4a_color_control_command_t command,
+                                                                  bool *support_auto,
+                                                                  int32_t *min_value,
+                                                                  int32_t *max_value,
+                                                                  int32_t *step_value,
+                                                                  int32_t *default_value,
```

Comment:
Can we have a functional test that reads these and confirms they are always the same for linux and windows?

We expect this to be consistent through out.

Reply:
Do you mean the values returned this function from Linux are same with from Windows?
Or, in certain platform, consecutive call of this function returns same values with previous call?

Reply:
after a device reset, I would expect the values returned by the device would be exactly the same. And as a result this API should be consistent between the two. Differences between UVC and MF should be fixed in code so they remain consistent.

---
In reply to: [280931778](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/pull/319#discussion_r280931778) [](ancestors = 280931778)

Reply:
That's good idea.
Just curious that because there is no guarantee that the functional test will be run after device reset, is it a good idea to reset the device in the functional test?
Do we have an API to reset the device?

Thanks.

Reply:
Added default value checking in the functional test with failure message if device was not reset after control value has changed.